pkg/controllers/machinesetsync/machineset_sync_controller_test.go

```diff
@@ -279,6 +307,50 @@ var _ = Describe("With a running MachineSetSync controller", func() {
 							))),
 					)
 				})
+
+				Context("when the MAPI machine set has a non-zero deletion timestamp", func() {
+					BeforeEach(func() {
+						Eventually(k.Object(mapiMachineSet), timeout).Should(
+							HaveField("ObjectMeta.Finalizers", ContainElement(machinesync.SyncFinalizer)),
+						)
+						Eventually(k.Object(capiMachineSet), timeout).Should(
+							HaveField("ObjectMeta.Finalizers", ContainElement(machinesync.SyncFinalizer)),
+						)
+						Expect(k8sClient.Delete(ctx, mapiMachineSet)).To(Succeed())
+					})
+					// Expect to see the finalizers, so they're in place before
+					//  we Expect logic that relies on them to work
+					It("should delete the CAPI machine set", func() {
+						// Does this need to be more specific, e.g explicitly be a 404?
+						// TODO: 404
+						Eventually(k.Get(capiMachineSet), timeout).ShouldNot(Succeed())
```

Comment:
You should be able to check the error perhaps with the `apierrors.IsNotFound` check, as a transform? Have a look at how `MatchError` works in gomega as an alternative